internal/pubsub/pubsub.go

```diff
@@ -197,7 +179,11 @@ func (s *Subscription) getNextBatch(ctx context.Context) error {
 			Body:     m.Body,
 			Metadata: m.Metadata,
 			ack: func() {
-				s.ackBatcher.Add(ackIDBox{id}, size)
+				s.wg.Add(1)
```

Comment:
@jba I find this pretty subtle; it's making the `ackBatcher.Add` non-blocking, but has to add the `WaitGroup` in order to synchronize the created goroutine with `Shutdown` later. WDYT about providing a non-blocking `Add` function in `driver.Batcher` instead (as `bundler.Bundler` did)? Something like `func AddNoWait(ctx context.Context, item interface{})`. IIUC, it would just mean running the final `select` in `Add` in a goroutine.

Reply:
OK.

Reply:
#807

Reply:
I'm not sure I understand how we can recover the functionality here using the new AddNoWait function from #807.

With what I have here, it's possible to wait for all these Adds to finish before shutdown. What's the solution using AddNoWait?

Reply:
I tried integrating this AddNoWait and got some breakage that I don't know how to fix. 
TestAckDoesNotBlock hung. I fixed it by using a time.Sleep call but I don't like it.
TestRetryTopic broke: "    pubsub_test.go:236: calls: got 0, want 3"

I was close to having something that worked, and I'd like to get it checked in before attempting this approach that I don't understand as well.